clients/src/main/java/org/apache/kafka/common/record/KafkaLZ4BlockInputStream.java

```diff
@@ -128,12 +129,14 @@ private void readHeader() throws IOException {
         header[headerOffset++] = (byte) in.read();
 
         // Old implementations produced incorrect HC checksums
-        if (ignoreHC)
+        if (ignoreFlagDescriptorChecksum)
             return;
 
         int offset = 4;
         int len = headerOffset - offset - 1; // dont include magic bytes or HC
-        assert len >= 2 && len <= 10 : "Checksum: Invalid FrameDescriptor size";
+        if (len < 2 || len > 10)
+            throw new AssertionError("Checksum: Invalid FrameDescriptor size");
```

Comment:
Is it worth keeping this? Seems like the kind of thing that should be verified via tests.